### PR TITLE
Adding isRetryable functionality ot BigtableAsyncRpc.

### PR DESCRIPTION
**Pull request by sduskis**

### CODE REVIEW

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java [hunk 1]
```diff
@@ -75,7 +72,8 @@
 public class BigtableDataGrpcClient implements BigtableDataClient {
 
   private static final Logger LOG = new Logger(BigtableDataGrpcClient.class);
-  private static AtomicInteger LOG_COUNT = new AtomicInteger();
+
+  /// Retryable Predicates
```
Comment (garye) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: Nit: /// -> //

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java [hunk 2]
```diff
@@ -115,11 +113,16 @@ public boolean apply(CheckAndMutateRowRequest checkAndMutateRowRequest) {
         }
       };
 
-  private static final Predicate<SampleRowKeysRequest> IS_RETRYABLE_SAMPLE_ROW_KEY =
-      Predicates.<SampleRowKeysRequest> alwaysTrue();
+  private static final boolean allCellsHaveTimestamps(Iterable<Mutation> mutations) {
+    for (Mutation mut : mutations) {
+      if (mut.getSetCell().getTimestampMicros() == -1) {
```
Comment (garye) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: What about if it's 0? That seems to be the default value in the Cell proto and what it gets set to after a clear.

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java (hunk 1 above)
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: fixed.

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java (hunk 2 above)
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java]: I changed this.  I hope that this doesn't bite some of our users who may have not set timestamp.

**Comment by garye** [comment]
LGTM

**Pull request merged**